db/db.go

```diff
@@ -11,66 +11,64 @@
 package db
 
 import (
-	"os"
+	"errors"
 
 	"github.com/Mierdin/todd/agent/defs"
 	"github.com/Mierdin/todd/config"
 	"github.com/Mierdin/todd/server/objects"
-	log "github.com/Sirupsen/logrus"
+)
+
+var (
+	ErrInvalidDBPlugin = errors.New("Invalid DB plugin in config file")
+	ErrNotExist        = errors.New("Value does not exist")
 )
 
 // DatabasePackage represents all of the behavior that a ToDD database plugin must support
 type DatabasePackage interface {
 
 	// (no args)
-	Init()
+	Init() error
 
 	// (agent advertisement to set)
-	SetAgent(defs.AgentAdvert)
+	SetAgent(defs.AgentAdvert) error
 
-	GetAgent(string) defs.AgentAdvert
-	GetAgents() []defs.AgentAdvert
+	GetAgent(string) (*defs.AgentAdvert, error)
```

Comment:
Silly me, I know better than this! You're absolutely right, there is no benefit to using a slice of pointers here, since the memory allocation is identical.